lib/bundler/plugin/index.rb

```diff
@@ -29,7 +29,12 @@ def initialize
         @hooks = {}
         @load_paths = {}
 
-        load_index(global_index_file, true)
+        begin
+          load_index(global_index_file, true)
+        rescue GenericSystemCallError
+          # no need to fail when on a read-only FS, for example
+          nil
```

Comment:
This should just rescue in the method, no need to have a `begin .. end` block.

Reply:
I'd prefer it being precise, this way nothing is caught by accident and it's more obvious why/what is being guarded, if the code ever needs moving nobody needs to go through the history and figure out what is being rescued

Reply:
fair enough